email_tracking/app.rb

```diff
@@ -0,0 +1,54 @@
+require 'sinatra/base'
+require 'easypost'
+require 'sendgrid-ruby'
+require 'tilt/erb'
+require 'pry'
+require 'dotenv'
+
+class App < Sinatra::Base
+  set :show_exceptions, :after_handler
+
+  configure :development, :test do
+    Dotenv.load
+  end
+
+  configure do
+    EasyPost.api_key = ENV['EASYPOST_API_KEY']
+    set :sendgrid, SendGrid::API.new(api_key: ENV['SENDGRID_API_KEY'])
+  end
+
+ post '/easypost-webhook' do
+    request.body.rewind
+    request_string = request.body.read
+    response = JSON.parse(request_string)
+
+    if response['object'] == 'Event' && response['description'] == 'tracker.updated'
+      event = EasyPost::Event.receive(request_string)
+      tracker = event.result
+
+      message = "Hey, this is FunCompany."
+      if tracker.status == 'delivered'
+        message += "Your package has arrived! "
+      else
+        message += "There's an update on your package: "
+      end
+
+      td = tracker.tracking_details.reverse.find{|tracking_detail| tracking_detail.status == tracker.status}
+      message += "%s says: %s in %s." % [tracker.carrier,tracking_detail.message,tracking_detail.tracking_location.city] if td.present?
```

Comment:
Use an interpolated string.

`"#{tracker.carrier} says: #{tracking_detail.message} in #{tracking_detail.tracking_location.city}."`